File: Threads Posix/exo2.c
#include <stdio.h>
#include <stdlib.h>
#include <pthread.h>
#include <time.h>

#define TAILLE 10

void* generer_tableau(void* arg) 
{
    
    int* tab = malloc(TAILLE * sizeof(int));
    
    srand(time(NULL));
    for (int i = 0; i < TAILLE; i++) 
    {
        tab[i] = rand() % 100;
    }
    
    
    return (void*)tab;
}

int main() 
{
    pthread_t thread;
    int* resultat;

    pthread_create(&thread, NULL, generer_tableau, NULL);

   
    pthread_join(thread, (void**)&resultat);

    printf("Tableau reçu du thread : ");
    for (int i = 0; i < TAILLE; i++) 
    {
        printf("%d ", resultat[i]);
    }
    printf("\n");

    free(resultat);
    return 0;
}

File: Threads Posix/exo2.py
import threading
import random
import time

def generer_tableau():
    tab = [random.randint(0, 99) for _ in range(10)]
    return tab

class ThreadAvecRetour(threading.Thread):
    def __init__(self, target):
        super().__init__()
        self.target = target
        self.resultat = None

    def run(self):
        self.resultat = self.target()

def main():
    thread = ThreadAvecRetour(target=generer_tableau)
    
    thread.start()
    thread.join()

    resultat = thread.resultat
    
    print("Tableau reçu du thread : ", end="")
    for val in resultat:
        print(f"{val} ", end="")
    print()

if __name__ == "__main__":
    main()